File: src/frontend/src/app/components/atendimento/tabela/model/tabela-grid-dto.ts
import { Exclude, Expose } from 'class-transformer';
import { TipoTabela } from './tipo-tabela.enum';

@Exclude()
export class TabelaGridDTO {
  @Expose() id?: string;
  @Expose() nome?: string;
  @Expose() tipo?: TipoTabela;
  @Expose() ativo?: boolean;
  @Expose() createdAt?: Date;
  @Expose() deleted?: boolean;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/tabela/dto/TabelaGridDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.tabela.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.atendimento.tabela.TipoTabela;
import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Getter
@Setter
public class TabelaGridDTO implements GridDTO {

    private UUID id;
    private String nome;
    private TipoTabela tipo;
    private Boolean ativo;
    private LocalDateTime createdAt;
    private Boolean deleted;
}
